src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,112 @@
 import React from 'react';
 import './TodoList.scss';
+import { Todo } from '../../react-app-env';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+type Props = {
+  todos: Todo[],
+  selectUser: (userId: number) => void,
+  selectedUserId: number,
+};
+
+type State = {
+  title: string,
+  visibleTodos: Todo[],
+};
+
+export class TodoList extends React.Component<Props, State> {
+  state = {
+    title: '',
+    visibleTodos: this.props.todos,
+  };
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.todos !== this.props.todos) {
+      this.state.visibleTodos = this.props.todos;
+    }
+  }
+
+  handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      title: event.target.value,
+    });
+  };
+
+  handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    switch (value) {
+      case ('active'):
+        this.setState(({
+          visibleTodos: this.props.todos.filter(todo => !todo.completed),
+        }));
```

Comment:
You can move it to external method so you won't repeat yourself in other cases, you can just pass filtering callback to it